controllers/replace_misconfigured_pods.go

```diff
@@ -78,22 +64,8 @@ func (c ReplaceMisconfiguredPods) Reconcile(r *FoundationDBClusterReconciler, co
 			continue
 		}
 
-		_, idNum, err := ParseInstanceID(instanceID)
-		if err != nil {
-			return &Requeue{Error: err}
-		}
-
-		processClass := internal.GetProcessClassFromMeta(pvc.ObjectMeta)
-		desiredPVC, err := internal.GetPvc(cluster, processClass, idNum)
-		if err != nil {
-			return &Requeue{Error: err}
-		}
-		pvcHash, err := internal.GetJSONHash(desiredPVC.Spec)
-		if err != nil {
-			return &Requeue{Error: err}
-		}
-
-		if pvc.Annotations[fdbtypes.LastSpecKey] != pvcHash {
+		hasNewRemovals, err = instanceNeedsRemovalForPVC(pvc, cluster)
+		if  hasNewRemovals {
```

Comment:
```suggestion
		if  needsNewRemoval {
```

Otherwise we overwrite `hasNewRemovals` every iteration.